tests/integration/event_filters_test.go

```diff
@@ -2189,14 +2158,44 @@ func ExpectAtLeastOneForEach(t *testing.T, cmdEvents []cmdEvents, actual *eventB
 						}
 					}
 				}
-
 				// if we got here, it means we found a match and can stop searching
-				found = true
-				break
+				return true, nil
+			}
+			return false, nil
+		}
+
+		// second stage: validate events
```

Comment:
I have added support for unexpected events only in this function (`ExpectAtLeastOneForEach`) as I only used this one.
If its seems good to you @NDStrahilevitz I can add it to all other functions as well.
I admit that I think that the current API created by @geyslan is a bit too complicated - it has too many functions so the differences between them is hard to understand. Anyways this PR is not about the testing infrastructure, so I only open it here for discussion.

Reply:
> I admit that I think that the current API created by @geyslan is a bit too complicated

I agree with you @AlonZivony. Integration API must be rethought.